project/profiles/capacity4more/behat/features/group_content_permissions.feature

```diff
@@ -14,109 +14,859 @@ Feature: Test creation of the content permissions.
+    When   I change access of group "Draft group" to "Private"
+    Then   I am an anonymous user
+    And    I go to "draft"
+    And    I should see "Please log in"
+
+  @api
+  Scenario Outline: As member, group admin, group owner and site admin, access create content form, with purl prefix of draft private group.
+    Given  I am logged in as user "<user>"
+    When   I go to "<path>"
+    Then   I should have access to the page
+
+    Examples:
+      | user        | path                 |
+      | isaacnewton | /draft/node/add/photoalbum |
+      | isaacnewton | /published/node/add/photo |
+      | isaacnewton | /published/node/add/event |
+      | isaacnewton | /published/node/add/document |
+      | isaacnewton | /published/node/add/discussion |
+      | turing      | /published/node/add/photoalbum |
+      | turing      | /published/node/add/photo |
+      | turing      | /published/node/add/event |
+      | turing      | /published/node/add/document |
+      | turing      | /published/node/add/discussion |
+      | turing      | /published/node/add/wiki-page |
+      | alfrednobel  | /published/node/add/photoalbum |
+      | alfrednobel  | /published/node/add/photo |
+      | alfrednobel  | /published/node/add/event |
+      | alfrednobel  | /published/node/add/document |
+      | alfrednobel  | /published/node/add/discussion |
+      | alfrednobel  | /published/node/add/wiki-page |
+      | mariecurie  | /published/node/add/photoalbum |
+      | mariecurie  | /published/node/add/photo |
+      | mariecurie  | /published/node/add/event |
+      | mariecurie  | /published/node/add/document |
+      | mariecurie  | /published/node/add/discussion |
+      | mariecurie  | /published/node/add/wiki-page |
+
+  @api
+  Scenario Outline: As non member, member (for wiki-page only) access create content form, with purl prefix of draft private group.
+    Given  I am logged in as user "<user>"
+    When   I go to "<path>"
+    Then   I should not have access to the page
+
+    Examples:
+      | user        | path                 |
+      | badhairday  | /draft/node/add/photoalbum |
+      | badhairday  | /draft/node/add/photo |
+      | badhairday  | /draft/node/add/event |
+      | badhairday  | /draft/node/add/document |
+      | badhairday  | /draft/node/add/discussion |
+      | badhairday  | /draft/node/add/wiki-page |
+      | isaacnewton | /draft/node/add/wiki-page |
+
+  @api
+  Scenario: Restore "Draft group" group state to 'Public'.
+    Given  I am logged in as user "mariecurie"
+    When   I change access of group "Draft group" to "Public"
+    And   I am an anonymous user
+
+  @api
+  Scenario: Set "Published group" group state to 'Public'. Anonymous user asked to log in.
+    Given  I am logged in as user "mariecurie"
+    When   I change access of group "Published group" to "Public"
+    Then   I am an anonymous user
+    And    I go to "published"
+    And    I should see "Please log in"
```

Comment:
wrong - a visitor can access and view all contents in a published public group - he does not need to login